src/main/java/com/atlassian/jira/cloud/jenkins/exceptions/JiraConnectionFailedException.java

```diff
@@ -0,0 +1,7 @@
+package com.atlassian.jira.cloud.jenkins.exceptions;
+
+public class JiraConnectionFailedException extends Exception {
+    public JiraConnectionFailedException(final String message) {
```

Comment:
Can we throw in some overloading? Might be good to have one that accounts for the cause and another that accounts for both:

```
public JiraConnectionFailedException(Throwable cause) {
    super(cause);
}

public JiraConnectionFailedException(String message, Throwable cause) {
    super(message, cause);
}
```